javascript/modules/client/src/internal/modules/graphql.ts

```diff
@@ -0,0 +1,82 @@
+import { Random } from "@aragon/sdk-common";
+import { GraphQLClient } from "graphql-request";
+import { Context } from "../../context";
+import { IClientGraphQLCore } from "../interfaces/core";
+import { QueryStatus } from "../graphql-queries";
+
+export class GraphqlModule implements IClientGraphQLCore {
+
+    private _graphql: GraphQLClient[] = [];
+    private _graphqlIdx: number = -1;
+
+    constructor(context: Context) {
+        if (context.graphql?.length) {
+            this._graphql = context.graphql
+            this._graphqlIdx = Math.floor(Random.getFloat() * context.graphql.length);
+        }
+    }
+
+    /**
+    * Get the current graphql client
+    * without any additional checks
+    * @returns {GraphQLClient}
+    */
+    public getClient(): GraphQLClient {
+        if (!this._graphql[this._graphqlIdx]) {
+            throw new Error("No graphql endpoints available");
+        }
+        return this._graphql[this._graphqlIdx];
+    }
+
+    /**
+     * Starts using the next available IPFS endpoint
+     * @returns {void}
+     */
+    public shiftClient(): void {
+        if (!this._graphql?.length) {
+            throw new Error("No graphql endpoints available");
+        } else if (this._graphql?.length < 2) {
+            throw new Error("No other endpoints");
+        }
+        this._graphqlIdx = (this._graphqlIdx + 1) % this._graphql?.length;
+    }
+
+    /**
+     * Checks if the current node is online
+     * @returns {Promise<boolean>}
+     */
+    public async isUp(): Promise<boolean> {
+        try {
+            const res = await this.getClient().request(QueryStatus)
+            if (res._meta?.deployment) {
+                return true
+            }
+            return false
+        } catch (e) {
+            return false
+        }
```

Comment:
Turning this into an `async` function creates a bigger bundle and overhead. If there is no specific need for `async/await` (if/else clauses, for await loops, etc) I'd suggest keeping it as it was.